src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Extensions/ApplicationInsightsServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,227 @@
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Logging.ApplicationInsights;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Configuration;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    internal static class ApplicationInsightsServiceCollectionExtensions
+    {
+        public static IServiceCollection AddApplicationInsights(this IServiceCollection services, Action<ApplicationInsightsLoggerOptions> configure)
+        {
+            services.AddApplicationInsights();
+            if (configure != null)
+            {
+                services.Configure<ApplicationInsightsLoggerOptions>(configure);
+            }
+            return services;
+        }
+
+        public static IServiceCollection AddApplicationInsights(this IServiceCollection services)
+        {
+            // Bind to the configuration section registered with 
+            services.AddOptions<ApplicationInsightsLoggerOptions>()
+                .Configure<ILoggerProviderConfiguration<ApplicationInsightsLoggerProvider>>((options, config) =>
+                {
+                    config.Configuration?.Bind(options);
+                });
+
+            services.AddSingleton<ITelemetryInitializer, HttpDependenciesParsingTelemetryInitializer>();
+            services.AddSingleton<ITelemetryInitializer, WebJobsRoleEnvironmentTelemetryInitializer>();
+            services.AddSingleton<ITelemetryInitializer, WebJobsTelemetryInitializer>();
+            services.AddSingleton<ITelemetryInitializer, WebJobsSanitizingInitializer>();
+            services.AddSingleton<ITelemetryModule, QuickPulseTelemetryModule>();
+            services.AddSingleton<ITelemetryModule, DependencyTrackingTelemetryModule>(provider =>
+            {
+                var dependencyCollector = new DependencyTrackingTelemetryModule();
+                var excludedDomains = dependencyCollector.ExcludeComponentCorrelationHttpHeadersOnDomains;
+                excludedDomains.Add("core.windows.net");
+                excludedDomains.Add("core.chinacloudapi.cn");
+                excludedDomains.Add("core.cloudapi.de");
+                excludedDomains.Add("core.usgovcloudapi.net");
+                excludedDomains.Add("localhost");
+                excludedDomains.Add("127.0.0.1");
+
+                return dependencyCollector;
+            });
+            services.AddSingleton<ITelemetryModule, AppServicesHeartbeatTelemetryModule>();
+
+            ServerTelemetryChannel serverChannel = new ServerTelemetryChannel();
+            services.AddSingleton<ITelemetryChannel>(serverChannel);
+            services.AddSingleton<TelemetryConfiguration>(provider =>
+            {
+                ApplicationInsightsLoggerOptions options = provider.GetService<IOptions<ApplicationInsightsLoggerOptions>>().Value;
```

Comment:
Would it make sense to wrap this logic up in a factory?

Reply:
Yes, I was going to bring that up with @lmolkova as well. It's getting pretty big and may be nice to split out. But not as part of this change.